Sources/ProjectSpec/SpecFile.swift

```diff
@@ -82,9 +82,12 @@ public struct SpecFile {
         }
     }
 
-    public func resolvedDictionary() -> JSONDictionary {
-        let resolvedSpec = resolvingPaths()
-        return resolvedSpec.mergedDictionary()
+    public func resolvedDictionary(environmentVariables: [String: String] = ProcessInfo.processInfo.environment) -> JSONDictionary {
```

Comment:
```suggestion
    public func resolvedDictionary(variables: [String: String] = [:]) -> JSONDictionary {
```
This lives in the project spec so we can allow it to handle variables coming from anywhere.